### PR TITLE
InventoryInfo stream import fix

### PR DESCRIPTION
**Pull request by yecli**
Returned InventoryInfo truly stream import, changed order of export to avoid error.

### CODE REVIEW

File: VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs [hunk 1]
```diff
@@ -117,24 +132,13 @@ public void DoImport(Stream backupStream, Action<ExportImportProgressInfo> progr
                             var fulfillmentCentersType = AbstractTypeFactory<FulfillmentCenter>.TryCreateInstance().GetType().MakeArrayType();
                             var fulfillmentCenters = _jsonSerializer.Deserialize(reader, fulfillmentCentersType) as FulfillmentCenter[];
 
-                            progressInfo.Description = $"The {fulfillmentCenters.Count()} fulfillmentCenters have been imported";
+                            progressInfo.Description = $"The {fulfillmentCenters.Count()} fulfillmentCenters has been imported";
```
Comment (artem-dudarev) [inline VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs]: `have`

File: VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs [hunk 2]
```diff
@@ -98,14 +98,29 @@ public void DoImport(Stream backupStream, Action<ExportImportProgressInfo> progr
...
                                     inventoryInfo = _jsonSerializer.Deserialize(reader, inventoryInfo.GetType()) as InventoryInfo;
-                                    inventoryInfos.Add(inventoryInfo);
+
+                                    inventoryInfoChunk.Add(inventoryInfo);
 
                                     reader.Read();
+
+                                    if (inventoryInfoChunk.Count >= BatchSize || reader.TokenType == JsonToken.EndArray)
+                                    {
+
```
Comment (artem-dudarev) [inline VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs]: Remove

File: VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs [hunk 3]
```diff
@@ -19,8 +19,9 @@ public BackupObject()
             InventoryInfos = Array.Empty<InventoryInfo>();
             FulfillmentCenters = Array.Empty<FulfillmentCenter>();
         }
-        public InventoryInfo[] InventoryInfos { get; set; }
+        // Important to set such property order as FulfillmentCenters should be saved before InventoryInfos when importing
```
Comment (artem-dudarev) [inline VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs]: This comment is unclear

File: VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs (hunk 1 above)
Comment (yecli) [reply VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs]: Done, in fact it was just revert to previous version.

File: VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs [hunk 4]
```diff
@@ -98,14 +98,28 @@ public void DoImport(Stream backupStream, Action<ExportImportProgressInfo> progr
...
-                                    inventoryInfos.Add(inventoryInfo);
+
+                                    inventoryInfoChunk.Add(inventoryInfo);
 
                                     reader.Read();
+
+                                    if (inventoryInfoChunk.Count >= BatchSize || reader.TokenType == JsonToken.EndArray)
+                                    {
+                                        progressInfo.ProcessedCount += inventoryInfoChunk.Count;
+                                        progressInfo.Description = $"{progressInfo.ProcessedCount} inventories records have been imported";
```
Comment (artem-dudarev) [inline VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs]: `records have been imported` means that the progress should be updated _after_ importing.

Comment (yecli) [reply VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs]: Changed.

**Pull request merged**